internal/testtypes/builders.go

```diff
@@ -0,0 +1,525 @@
+type ClusterClassBuilder struct {
+	namespace                                 string
+	name                                      string
+	infrastructureClusterTemplate             *unstructured.Unstructured
+	controlPlaneMetadata                      *clusterv1.ObjectMeta
+	controlPlaneTemplate                      *unstructured.Unstructured
+	controlPlaneInfrastructureMachineTemplate *unstructured.Unstructured
+	machineDeploymentClasses                  []clusterv1.MachineDeploymentClass
+}
+
+// NewClusterClassBuilder returns a ClusterClassBuilder with the given name and namespace.
+func NewClusterClassBuilder(namespace, name string) *ClusterClassBuilder {
+	return &ClusterClassBuilder{
+		namespace: namespace,
+		name:      name,
+	}
+}
+
+// WithInfrastructureClusterTemplate adds the passed InfrastructureClusterTemplate to the ClusterClassBuilder.
+func (c *ClusterClassBuilder) WithInfrastructureClusterTemplate(t *unstructured.Unstructured) *ClusterClassBuilder {
+	c.infrastructureClusterTemplate = t
+	return c
+}
+
+// WithControlPlaneTemplate adds the passed ControlPlaneTemplate to the ClusterClassBuilder.
+func (c *ClusterClassBuilder) WithControlPlaneTemplate(t *unstructured.Unstructured) *ClusterClassBuilder {
+	c.controlPlaneTemplate = t
+	return c
+}
+
+// WithControlPlaneMetadata adds the given labels and annotations for use with the ControlPlane to the ClusterClassBuilder.
+func (c *ClusterClassBuilder) WithControlPlaneMetadata(labels, annotations map[string]string) *ClusterClassBuilder {
+	c.controlPlaneMetadata = &clusterv1.ObjectMeta{
+		Labels:      labels,
+		Annotations: annotations,
+	}
+	return c
+}
+
+// WithControlPlaneInfrastructureMachineTemplate adds the ControlPlane's InfrastructureMachineTemplate to the ClusterClassBuilder.
+func (c *ClusterClassBuilder) WithControlPlaneInfrastructureMachineTemplate(t *unstructured.Unstructured) *ClusterClassBuilder {
+	c.controlPlaneInfrastructureMachineTemplate = t
+	return c
+}
+
+// WithWorkerMachineDeploymentClasses adds the variables and objects needed to create MachineDeploymentTemplates for a ClusterClassBuilder.
+func (c *ClusterClassBuilder) WithWorkerMachineDeploymentClasses(mdcs []clusterv1.MachineDeploymentClass) *ClusterClassBuilder {
+	if c.machineDeploymentClasses == nil {
+		c.machineDeploymentClasses = make([]clusterv1.MachineDeploymentClass, 0)
+	}
+	c.machineDeploymentClasses = append(c.machineDeploymentClasses, mdcs...)
+	return c
+}
+
+// Build takes the objects and variables in the ClusterClass builder and uses them to create a ClusterClass object.
+func (c *ClusterClassBuilder) Build() *clusterv1.ClusterClass {
+	obj := &clusterv1.ClusterClass{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ClusterClass",
+			APIVersion: clusterv1.GroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      c.name,
+			Namespace: c.namespace,
+		},
+		Spec: clusterv1.ClusterClassSpec{},
```

Comment:
```suggestion
```
nit, if you want. As ClusterClassSpec is not a pointer you don't have to set it.

Reply:
I prefer the explicitness of this, so I'd prefer to leave it for now. I think it makes it easier to see the shape and structure of the object.